validate_yaml.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+"""
+Validates all yaml files by testing if entities and intents
+are unique and if all files mentioned are present
+"""
+
+import glob
+import os
+import sys
+
+import yaml
+from fire import Fire
+
+from validate_regex import validate_regex_in_yaml_file
+
+
+def print_or_throw_error(condition, success_message, failure_message, code=1):
+    "Print a message on success/failure of condition"
+    if condition:
+        print(success_message)
+    else:
+        assert False, failure_message
+
+
+def validate_elements(new_elements, known_elements):
+    """
+    Check new elements (entities or intents) against already identified elements.
+    Returns elment list, success_code where
+    element list is the bad entities if it failed
+    >>> validate_elements([1,2,3], [])
+    ([1, 2, 3], True)
+    >>> validate_elements([1,2,2], [])
+    ([2], False)
+    >>> validate_elements([], [])
+    ([], True)
+    >>> validate_elements([1,2,2], [2])
+    ([2], False)
+    >>> validate_elements([1,2], [2])
+    ([2], False)
+    """
+    unique_elements = set(new_elements)
+    is_unique = sorted(unique_elements) == sorted(new_elements)
+
+    overlap = unique_elements.intersection(known_elements)
+    is_known = bool(overlap)
+
+    if not is_unique:
+        duplicated_elements = set(elem for elem in new_elements
+                                  if new_elements.count(elem) > 1)
+        overlap = overlap.union(duplicated_elements)
+    failure = is_known or not is_unique
+    return (list(overlap if failure else unique_elements), not failure)
+
+
+def validate_file_exists(file_name):
+    "This wrapper checks if a file exists."
+    if not os.path.isfile(file_name):
+        throw_error("{} does not exist".format(file_name))
```

Comment:
We could be a little more specific.
```suggestion
        throw_error("{} file does not exist".format(file_name))
```